src/rabbit_mgmt_wm_definitions.erl

```diff
@@ -426,3 +437,71 @@ rv(VHost, Type, Props) -> rv(VHost, Type, name, Props).
 
 rv(VHost, Type, Name, Props) ->
     rabbit_misc:r(VHost, Type, maps:get(Name, Props, undefined)).
+
+%%--------------------------------------------------------------------
+
+validate_limits(All) ->
+    case maps:get(queues, All, undefined) of
+        undefined -> ok;
+        Queues0 ->
+            {ok, VHostMap} = filter_out_existing_queues(Queues0),
+            maps:fold(fun validate_vhost_limit/3, ok, VHostMap)
+    end.
+
+validate_limits(All, VHost) ->
+    case maps:get(queues, All, undefined) of
+        undefined -> ok;
+        Queues0 ->
+            Queues1 = filter_out_existing_queues(VHost, Queues0),
+            AddCount = length(Queues1),
+            validate_vhost_limit(VHost, AddCount, ok)
+    end.
+
+filter_out_existing_queues(Queues) ->
+    QData = [ build_queue_data(Q) || Q <- Queues ],
```

Comment:
I don't think is going to be a problem for the import, but it is possible to avoid two passes through the queue list (list comprehension here and `build_filtered_map` just after) by calling `build_queue_data` inside `build_filtered_map` for every queue.  The same applies to `filter_out_existing_queues/2`